File: frontend/src/api/consent.js
import client from "./client";

/**
 * POST /consent/
 * Backend schema (ConsentRequest) now accepts the four real UI toggles:
 *   employment_signals_consent, professional_presence_consent,
 *   digital_signals_consent, public_data_consent
 * plus the three legacy buckets (digital/transaction/behavioral) that the
 * older monitoring/reporting layer still expects.
 */
export function saveConsent({
  customerId,
  employmentSignals,
  professionalPresence,
  digitalSignals,
  publicData,
  version = "1.0",
}) {
  const payload = {
    customer_id: customerId,
    digital_data_consent: digitalSignals,
    transaction_data_consent: publicData,
    behavioral_data_consent: employmentSignals || professionalPresence,
    employment_signals_consent: employmentSignals,
    professional_presence_consent: professionalPresence,
    digital_signals_consent: digitalSignals,
    public_data_consent: publicData,
    version,
  };
  return client.post("/consent/", payload).then((res) => res.data);
}

File: backend/app/schemas/consent.py
from pydantic import BaseModel

from typing import Optional


class ConsentCreate(BaseModel):

    customer_id: int

    digital_data_consent: bool

    transaction_data_consent: bool

    behavioral_data_consent: bool

    version: str = "1.0"


class ConsentRequest(ConsentCreate):

    # The four toggles collected on the Consent Manager screen, kept
    # optional for backward compatibility with older clients.
    employment_signals_consent: Optional[bool] = None

    professional_presence_consent: Optional[bool] = None

    digital_signals_consent: Optional[bool] = None

    public_data_consent: Optional[bool] = None


class ConsentResponse(ConsentCreate):

    id: int

    employment_signals_consent: Optional[bool] = None

    professional_presence_consent: Optional[bool] = None

    digital_signals_consent: Optional[bool] = None

    public_data_consent: Optional[bool] = None

    class Config:
        from_attributes = True